File: leetcode/dp/1d/0198_house_robber.py
# LC 198. House Robber | Medium
class Solution:
    def rob(self, nums):
        if not nums: return 0
        prev2, prev1 = 0, 0
        for n in nums: prev2, prev1 = prev1, max(prev1, prev2 + n)
        return prev1

File: leetcode/dp/1d/0198_house_robber.cpp
// LC 198. House Robber | Medium
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int rob(vector<int>& nums) {
        int p2 = 0, p1 = 0;
        for (int n : nums) { int t = p1; p1 = max(p1, p2 + n); p2 = t; }
        return p1;
    }
};
